rust/src/lib.rs

```diff
@@ -20,25 +20,27 @@ mod ffi {
     // TODO(gitbuda): Having input Element object under ffi is a problem for general solution.
     // NOTE: This struct is / should be aligned with the schema.
     struct Element {
+        data: String,
```

Comment:
`ExternalStoreMock` is source of truth here, split data into `metadata` and `data`